slasher/beaconclient/historical_data_retrieval.go

```diff
@@ -21,6 +21,12 @@ func (bs *Service) RequestHistoricalAttestations(
 	res := &ethpb.ListIndexedAttestationsResponse{}
 	var err error
 	for {
+		if ctx.Err() == context.Canceled {
```

Comment:
We can make this more generic with if ctx.Err() != nil

Reply:
Thanks, fixed.